app-server/app/src/main/scala/user/Routes.scala

```diff
@@ -22,49 +19,53 @@ trait UserRoutes extends Authentication with PaginationDirectives {
   implicit val database: Database
   implicit val ec: ExecutionContext
 
+  val userExceptionHandler = ExceptionHandler {
+    case e: UserErrorException =>
+      complete(StatusCodes.ClientError(400)("", e.getMessage()))
+  }
+
   def userRoutes:Route = {
-    authenticate { user =>
-      pathPrefix("api" / "users") {
-        pathEndOrSingleSlash {
-          withPagination { page =>
-            get {
-              onSuccess(UserService.getPaginatedUsers(page)) { resp =>
-                complete(resp)
-              }
-            }
-          } ~
-          post {
-            entity(as[UsersRowCreate]) { newUser =>
-              onSuccess(UserService.createUser(newUser)) {
-                case Success(user) => complete((StatusCodes.Created, user))
-                case Failure(_) => complete(StatusCodes.InternalServerError)
-              }
-            }
-          }
-        } ~
-        pathPrefix(JavaUUID) { id =>
+    handleExceptions(userExceptionHandler) {
+      authenticate { user =>
+        pathPrefix("api" / "users") {
           pathEndOrSingleSlash {
-            get {
-              onSuccess(UserService.getUserById(id)) { resp =>
-                resp match {
-                  case Some(user) => complete(user)
-                  case _ => complete((StatusCodes.NotFound))
+            withPagination { page =>
+              get {
+                onSuccess(UserService.getPaginatedUsers(page)) { resp =>
+                  complete(resp)
                 }
               }
             } ~
-            put {
-              entity(as[UsersRow]) { user =>
-                onSuccess(UserService.updateUser(user, id)) { resp =>
-                  resp match {
-                    case Success(code) => code match {
-                      case 1 => complete((StatusCodes.NoContent))
-                      case 0 => complete((StatusCodes.NotFound))
+              post {
+                //TODO: This should only be accessible by users with the correct permission(IE admin in the "Public" org)
+                entity(as[UsersRowCreate]) { newUser =>
+                  onSuccess(UserService.createUser(newUser)) {
+                    case Success(user) => onSuccess(UserService.getUserWithOrgsById(user.id)) {
+                      case Some(user) => complete((StatusCodes.Created, user))
+                      case None => complete(StatusCodes.InternalServerError)
                     }
                     case Failure(_) => complete(StatusCodes.InternalServerError)
                   }
                 }
               }
+          } ~
+            pathPrefix(Segment) {
```

Comment:
I'm kind of torn about this - I think this is a little confusing though the way this single route is split up and the rest aren't.

I think this would work instead:

```scala
pathPrefix(Segment) { authIdEncoded =>
          val authId = java.net.URLDecoder.decode(authIdEncoded, "US-ASCII")
          pathEndOrSingleSlash {
            get {
              onSuccess(UserService.getUserWithOrgsById(authId)) { resp =>
                resp match {
                  case Some(user) => complete(user)
                  case _ => complete((StatusCodes.NotFound))
                }
              }
            }
          }
        }```

Also, is `US-ASCII` correct since URLs can contain utf-8 encoded strings?